cmds/portmaster-start/lock.go

```diff
@@ -32,14 +32,42 @@ func checkAndCreateInstanceLock(name string) (pid int32, err error) {
 		return 0, createInstanceLock(lockFilePath)
 	}
 
-	// check if process exists
+	// Check if process exists.
 	p, err := processInfo.NewProcess(int32(parsedPid))
-	if err == nil {
-		return p.Pid, nil
+	if err != nil {
+		// A process with the locked PID does not exist.
+		// This is expected, so we can continue normally.
+		return 0, createInstanceLock(lockFilePath)
+	}
+
+	// Get the process paths and evaluate and clean them.
+	executingBinaryPath, err := p.Exe()
+	if err != nil {
+		return 0, fmt.Errorf("failed to get path of existing process: %w", err)
+	}
+	cleanedExecutingBinaryPath, err := filepath.EvalSymlinks(executingBinaryPath)
+	if err != nil {
+		return 0, fmt.Errorf("failed to evaluate path of existing process: %w", err)
+	}
+	ownBinaryPath, err := os.Executable()
+	if err != nil {
+		return 0, fmt.Errorf("failed to get path of own process: %w", err)
+	}
+	cleanedOwnBinaryPath, err := filepath.EvalSymlinks(ownBinaryPath)
+	if err != nil {
+		return 0, fmt.Errorf("failed to evaluate path of own process: %w", err)
+	}
+
+	// Check if the binary path matches.
+	if cleanedExecutingBinaryPath != cleanedOwnBinaryPath {
+		// The process with the locked PID belongs to another binary.
+		// As the Portmaster usually starts very early, it will have a low PID,
+		// which could be assigned to another process on next boot.
+		return 0, createInstanceLock(lockFilePath)
 	}
```

Comment:
I would remove the part with the "low PID" because it doesn't really matter. If we failed to cleanup our PID file we may have the situtation even if the PID is very high

Reply:
We have experienced this issue on both Linux and Windows now. I think we already do everything right with the cleaning up. But apparently, that is not enough.